### PR TITLE
New docs for Boostraps and Relays

### PR DESCRIPTION
**Pull request by momosh-ethernal**
Docs that could serve as a deployment guideline for Bootstrap and Relay nodes.

### CODE REVIEW

File: docs/join-the-network/node-types.md [hunk 1]
```diff
@@ -82,3 +82,35 @@ network.
 
 The RPC node serves as a gateway for developers and users to interact
 with the Avail network without the need to run a full node locally.
+
+### Bootstrap nodes
```
Comment (prabal-banerjee) [inline docs/join-the-network/node-types.md:86]: Bcoz this is relevant for both Substrate and LC networks, maybe we should add a line that both networks require this type of node?

File: docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  - avail
+  - node
+  - data availability
+  - da
+image: https://availproject.github.io/img/avail/AvailDocs.png
+slug: avail-bootstrap-node
+---
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+## Bootstrap Node Setup
```
Comment (prabal-banerjee) [inline docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md:17]: We need to clarify that this is for LC and not the Substrate node - it might create confusion.

File: docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+touch config.yaml
+nano config.yaml
+
+```
+
+You can use our example file:
+```yaml
+log_level = 'INFO'
+log_format_json = false
+secret_key = { seed = "0" }
```
Comment (prabal-banerjee) [inline docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md:54]: Good to remove this, otherwise this will create conflict with other nodes?

File: docs/join-the-network/run-avail/other-nodes/0060-avail-relay-node.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  - data availability
+  - da
+image: https://availproject.github.io/img/avail/AvailDocs.png
+slug: avail-relay-node
+---
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+## Relay Node Setup
+
+If for some reason you ever wanted to host your own network, you will most likely encounter connectivity issues and difficulties between your netwokr peers.
```
Comment (prabal-banerjee) [inline docs/join-the-network/run-avail/other-nodes/0060-avail-relay-node.md:19]: typo: `netwokr` -> `network`

File: docs/join-the-network/run-avail/other-nodes/0060-avail-relay-node.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+touch config.yaml
+nano config.yaml
+
+```
+
+You can use our example file:
+```yaml
+log_level = 'INFO'
+log_format_json = false
+secret_key = { seed = "0" }
```
Comment (prabal-banerjee) [inline docs/join-the-network/run-avail/other-nodes/0060-avail-relay-node.md:54]: same as earlier

**Review by prabal-banerjee — APPROVED** [review]

File: docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  - da
+image: https://availproject.github.io/img/avail/AvailDocs.png
+slug: avail-bootstrap-node
+---
+import useBaseUrl from '@docusaurus/useBaseUrl';
+
+## Bootstrap Node Setup
+
+If for some reason you ever wanted to host your own network, you will need a node that will serve as an access point to all the other nodes wanting to join.
+This is where the Bootstrap node comes onto to the stage. We offer a simple and functioning setup for a such node, just be sure to deploy few of these and that they are publicly accessible over the network.
```
Comment (obrienalaribe) [inline docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md:20]: typo, maybe rephrase as "we offer a simple functioning bootnode setup" and maybe emphasize the importance of a bootnode i.e their role and why they need to be publicly accessible over a network. Perhaps we can give a guideline on how many bootnodes are sufficient to kickstart the network

File: docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md (hunk 2 above)
Comment (obrienalaribe) [reply docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md:17]: Agree with @prabal-banerjee perhaps something like this 

`bootstrap nodes play a crucial role in facilitating initial network discovery and connection establishment. By connecting to the bootstrap nodes, new nodes can discover and connect to other peers in the network, gradually expanding the network's connectivity`

File: docs/join-the-network/node-types.md [hunk 7]
```diff
@@ -82,3 +82,35 @@ network.
...
+these networks.
+
+After a Boot node had served it purpose as an entry point to the network,
+and had provided a list of other nodes in the network to the newcomers,
+new nodes will no longer relly on them.
+
+### Relay nodes
+There will certainly be situations were means of establishing a direct 
+communication between peers is unfeasible. These situations require the 
+assistance of a third peer willing and able to act as a intermediary.
```
Comment (obrienalaribe) [inline docs/join-the-network/node-types.md:103]: able to act as `an` intermediary

File: docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+First, binary for the Bootstrap node needs to be compiled. On your local machine, or either on any cloud provider VM, be sure that Rust is installed:
+```
+curl https://sh.rustup.rs -sSf | sh
+source $HOME/.cargo/env
+rustup update nightly
+rustup target add wasm32-unknown-unknown --toolchain nightly
+rustc --version # verify rust is working, print the installed version
+```
```
Comment (obrienalaribe) [inline docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md:29]: Might need to install some compiler packages if its a new ubuntu machine. I ran this `apt install build-essential`

File: docs/join-the-network/node-types.md (hunk 1 above)
Comment (momosh-ethernal) [reply docs/join-the-network/node-types.md:86]: Went for a wider description there, that is applicable in pretty much all general cases, not only our networks

File: docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md (hunk 3 above)
Comment (momosh-ethernal) [reply docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md:54]: If a repeatable multiaddress is needed, and in this case I think that answer is yes, then seeds for secret keys need to be controlled in a fashion where they are not repeated. 

That `0` there is just an example

Comment (DannyS03) [reply docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md]: changed to `secret_key: { seed: "unique_seed" }`

**Comment by DannyS03** [comment]
Addressed outstanding feedback and made enhancements. Will revisit very soon as part of efforts to update all How-Tos for improved context and cross-referencing.

File: docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md (hunk 3 above)
Comment (prabal-banerjee) [reply docs/join-the-network/run-avail/other-nodes/0050-avail-bootstrap-node.md]: Isn't keeping it empty mean that random seed will be generated?
